dll/win32/uxtheme/msstyles.c

```diff
@@ -893,6 +893,56 @@ static BOOL prepare_alpha (HBITMAP bmp, BOOL* hasAlpha)
     return TRUE;
 }
 
+#ifdef MSSTYLES_ACCEPT_PNG_IMAGES
+static
+BYTE
+process_png_color_channel(
+    _In_ BYTE value,
+    _In_ float a)
+{
+    if ((a > 0.0f) && (a < 1.0f))
+        return value / a;
+    else
+        return value;
+}
+
+BOOL
+prepare_png_alpha(
+    _In_ HBITMAP png,
+    _Out_ BOOL* hasAlpha)
+{
+    DIBSECTION dib;
+    int n;
+    BYTE* p;
+
+    *hasAlpha = FALSE;
+
+    if (!png || GetObjectW( png, sizeof(dib), &dib ) != sizeof(dib))
+        return FALSE;
+
+    if(dib.dsBm.bmBitsPixel != 32)
+        /* nothing to do */
+        return TRUE;
+
+    p = dib.dsBm.bmBits;
+    n = dib.dsBmih.biHeight * dib.dsBmih.biWidth;
+    while (n-- > 0)
+    {
+        float a = p[3] / 255.0f;
+        
+        p[0] = process_png_color_channel(p[0], a);
+        p[1] = process_png_color_channel(p[1], a);
+        p[2] = process_png_color_channel(p[2], a);
+        p += 4;
+        
+        if (a < 1.0f)
+            *hasAlpha = TRUE;
+    }
+
+    return TRUE;
+}
+#endif
```

Comment:
The closing comment here is more important than in other places, since the block contains more lines. You can omit this comment where a few lines are used.
```suggestion
#endif /* MSSTYLES_ACCEPT_PNG_IMAGES */
```